whois-api/src/main/java/net/ripe/db/whois/api/httpserver/dos/IpBlockListFilter.java

```diff
@@ -0,0 +1,66 @@
+package net.ripe.db.whois.api.httpserver.dos;
+
+import com.google.common.net.InetAddresses;
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.FilterConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import net.ripe.db.whois.common.Latin1Conversion;
+import net.ripe.db.whois.common.hazelcast.HazelcastBlockedIps;
+import net.ripe.db.whois.common.ip.IpInterval;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+
+@Component
+public class IpBlockListFilter implements Filter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IpBlockListFilter.class);
+
+    private final HazelcastBlockedIps hazelcastBlockedIps;
+
+    public IpBlockListFilter(final HazelcastBlockedIps hazelcastBlockedIps){
+        this.hazelcastBlockedIps = hazelcastBlockedIps;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain) throws IOException, ServletException {
```

Comment:
Is this filter gets applied after remoteAddress customizer and DosFilter?

Reply:
Before the DoSFilter is a requirement (read the implementation section of the task).
If the IP is permanent blocked shouldn't go to DoSFilter.

After the RemoteAddress yes, you can easily check this looking the ITs. We are using the clientIP there which is being moved to remoteAddress. You can easily check this as well putting a breakpoint in RemoteAddressFilter and running an IT.